evals/registry.py

```diff
@@ -82,7 +82,12 @@ def add_registry_paths(self, paths: list[Union[str, Path]]):
 
     @cached_property
     def api_model_ids(self):
-        return [m["id"] for m in openai.Model.list()["data"]]
+        try:
+            return [m["id"] for m in openai.Model.list()["data"]]
+        except openai.error.OpenAIError:
+            # Errors can happen when running eval with completion function that uses custom
+            # API endpoints and authentication mechanisms.
+            return []
```

Comment:
When the user is using an OpenAI model and encounters this error, the behavior will be misleading. Could you add a warning log here?

Reply:
Make sense, added.